commit 68d94bbe50b7577d48998107d632ef3a0df050e3
Author: Florian Weimer <fweimer@redhat.com>
Date:   Fri Aug 28 10:26:07 2026 +0200

    iconvdata: SHIFT_JISX0213 decoding lacks pending character reset (CVE-2026-77117)
    
    This fixes bug 34556.
    
    Reviewed-by: Carlos O'Donell <carlos@redhat.com>

diff --git a/iconvdata/shift_jisx0213.c b/iconvdata/shift_jisx0213.c
index 8876adc7d0b40c4f..c89d9986bc59a455 100644
--- a/iconvdata/shift_jisx0213.c
+++ b/iconvdata/shift_jisx0213.c
@@ -227,6 +227,9 @@
 	    STANDARD_FROM_LOOP_ERR_HANDLER (1);				      \
 	  }								      \
       }									      \
+    else 								      \
+      /* There was a pending character.  Clear it.  */			      \
+      *statep = 0;							      \
 									      \
     put32 (outptr, ch);							      \
     outptr += 4;							      \
